### PR TITLE
feat: TM sell buttons removed for crafteds and ID'd gear

### PR DESCRIPTION
**Pull request by Realswitzer**
Should have been part of the last PR in hindsight, but this fixes the sell all button for identified gear which can only be sold individually. Unidentified gear will work as normal.

Decided the cleanest place to put it was in `TradeMarketBulkSellFeature`, and materials get checked before due to being substantially more common, functionally very similar check.

### CODE REVIEW

**Comment by Realswitzer** [comment]
temporarily setting as draft because the button is gone? worked fine in initial testing

File: common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java [hunk 1]
```diff
@@ -388,6 +395,10 @@ private void handleSellDialogueUpdate() {
                     MaterialItem materialItem = materialItemOpt.get();
                     soldItemName = materialItem.getName();
                     soldItemTier = Optional.of(materialItem.getQualityTier());
+                } else if (Models.Item.asWynnItem(itemStack, GearItem.class).isPresent()) {
+                    if (!soldItemName.startsWith("Unidentified ")) {
+                        soldItemIdentified = Optional.of(true);
```
Comment (ryanzhoudev) [inline common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: Maybe possible to use GearItem.isUnidentified instead here

Comment (Realswitzer) [reply common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: it is possible, and i can make that change, but imo it is substantially messier due to the extra nesting (and i'm unsure about performance overhead, as insignificant as it would likely be)

```diff
diff --git a/common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java b/common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java
index b56a41257..d5ae56db9 100644
--- a/common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java
+++ b/common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java
@@ -395,9 +395,13 @@ public final class TradeMarketModel extends Model {
                     MaterialItem materialItem = materialItemOpt.get();
                     soldItemName = materialItem.getName();
                     soldItemTier = Optional.of(materialItem.getQualityTier());
-                } else if (Models.Item.asWynnItem(itemStack, GearItem.class).isPresent()) {
-                    if (!soldItemName.startsWith("Unidentified ")) {
-                        soldItemIdentified = Optional.of(true);
+                } else {
+                    Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+                    if (gearItemOpt.isPresent()) {
+                        GearItem gearItem = gearItemOpt.get();
+                        if (!gearItem.isUnidentified()) {
+                            soldItemIdentified = Optional.of(true);
+                        }
                     }
                 }
             }
```

**Comment by Realswitzer** [comment]
after having some time to think about it, maybe it would just be better to fully remove the sell buttons if only a single item can be sold (gears, crafteds, etc.)? it might be more intuitive than sell all resetting the count

**Comment by ryanzhoudev** [comment]
yeah good idea

**Comment by Realswitzer** [comment]
~for the moment, because i dont have potions on standby, multiple emerald pouches, etc, i am unable to fully test and remove buttons for other singular-only items, since there's some weird edge cases (potion of wisdom sell limit is 100, unsure if stat potions, mana potions, etc. have same limit)~

after a quick test, selling multiple potions and emerald pouches works fine, so this looks final unless theres any other edge cases

File: common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java [hunk 2]
```diff
@@ -123,6 +126,7 @@ public final class TradeMarketModel extends Model {
 
     private String soldItemName = null;
     private Optional<Integer> soldItemTier = Optional.empty();
+    private Optional<Boolean> disableSellButtons = Optional.empty();
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: Just makes this a boolean `shouldDisplaySellButtons` instead

File: common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java [hunk 3]
```diff
@@ -101,7 +103,8 @@ public final class TradeMarketModel extends Model {
     private static final Pattern PRICE_PATTERN = Pattern.compile(
             "§[67] - (?:§f(?<amount>[\\d,]+) §7x )?§(?:(?:(?:c✖|a✔) §f)|f§m|f)(?<price>[\\d,]+)§7(?:§m)?²(?:§b ✮ (?<silverbullPrice>[\\d,]+)§3²)?(?: .+)?");
 
-    private static final Pattern SELL_ITEM_NAME_PATTERN = Pattern.compile("\uDAFC\uDC00§.(.+)\uDAFC\uDC00");
+    // Test in TradeMarketModel_SELL_ITEM_NAME_PATTERN
+    private static final Pattern SELL_ITEM_NAME_PATTERN = Pattern.compile("(?:\uDAFC\uDC00|)§.(.+)(?:\uDAFC\uDC00|)");
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: Make the colour optional, it's not matching items with white names (Which are items that don't support sell all anyway but still)

Comment (Realswitzer) [reply common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: any items specifically? don't have a test case since i don't know about any that are missing the color
tried on white gear which matches, i cant find a thing that doesnt match the regex (white gear, quest items, horse whistle, etc)

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: Mount items were not matching for me, but those you can only sell 1 of even if same stats. But there might be an item in the future that does support it

Comment (Realswitzer) [reply common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java]: oh, i see. didnt notice since the buttons werent there (ib)
looking into it right now

**Comment by Realswitzer** [comment]
only issue with trying to account for items missing the item marker is the regex matches any ending item marker as part of $1, whereas the old pattern didnt. still matches, but i dont know how important it is to keep in line with the old implementation

File: common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java [hunk 4]
```diff
@@ -389,6 +399,15 @@ private void handleSellDialogueUpdate() {
...
                 }
+                Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+                if (gearItemOpt.isPresent() && !gearItemOpt.get().isUnidentified()) {
+                    shouldDisableSellButtons = true;
+                }
+                Optional<CraftedItemProperty> craftedItemPropertyOpt =
+                        Models.Item.asWynnItemProperty(itemStack, CraftedItemProperty.class);
+                if (craftedItemPropertyOpt.isPresent()) {
+                    shouldDisableSellButtons = true;
+                }
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/models/trademarket/TradeMarketModel.java:411]: You can add MountItem's to this as well, then I think this is good to go

**Review by ShadowCat117 — APPROVED** [review]
Looks good thanks!

**Review by kristofbolyai — APPROVED** [review]
